wrappers/node/demo/alice.js

```diff
@@ -99,22 +99,34 @@ async function runAlice (options) {
     requests = await DisclosedProof.getRequests(connectionToFaber)
   }
   logger.info('#23 Create a Disclosed proof object from proof request')
+  logger.info(`#23 Received proof request = ${JSON.stringify(requests)}`)
   const proof = await DisclosedProof.create({ sourceId: 'proof', request: JSON.stringify(requests[0]) })
 
   logger.info('#24 Query for credentials in the wallet that satisfy the proof request')
-  const credentials = await proof.getCredentials()
-
-  // Use the first available credentials to satisfy the proof request
-  for (let i = 0; i < Object.keys(credentials.attrs).length; i++) {
-    const attr = Object.keys(credentials.attrs)[i]
-    credentials.attrs[attr] = {
-      credential: credentials.attrs[attr][0]
+  const resolvedCreds = await proof.getCredentials()
+  const selectedCreds = {attrs : {}}
+  logger.info(`Resolved credentials for proof= ${JSON.stringify(resolvedCreds)}`)
+
+  for (const attrKey of Object.keys(resolvedCreds.attrs)) {
+    const attrCredInfo = resolvedCreds.attrs[attrKey]
+    logger.warn(`attrCredInfo for ${attrKey} = ${attrCredInfo}`)
+    if (Array.isArray(attrCredInfo) === false) {
+      throw Error(`Unexpected data, expected attrCredInfo to be array.`)
+    }
+    if (attrCredInfo.length > 0) {
+      selectedCreds.attrs[attrKey] = {
```

Comment:
I think it would be better to call `attrKey` an `attrName`, or keept `attr`, `attrKey` is less accurate and too generic.